src/lib/bootloader/cpu_mitigations.rb

```diff
@@ -0,0 +1,75 @@
+require "yast"
+
+require "cfa/matcher"
+require "cfa/placer"
+
+module Bootloader
+  # Specialized class to handle CPU mitigation settings.
+  # @see https://www.suse.com/support/kb/doc/?id=7023836
+  class CpuMitigations
+    include Yast::Logger
+    include Yast::I18n
+    extend Yast::I18n
+    KERNEL_MAPPING = {
+      nosmt:  "auto,nosmt",
+      auto:   "auto",
+      off:    "off",
+      manual: nil
+    }.freeze
+
+    HUMAN_MAPPING = {
+      nosmt:  N_("Auto + No SMT"),
+      auto:   N_("Auto"),
+      off:    N_("Off"),
+      manual: N_("Manually")
+    }.freeze
+
+    attr_reader :value
+
+    def initialize(value)
+      textdomain "bootloader"
+
+      @value = value
+    end
+
+    # Note: order of ALL is used also in UI as order of combobox.
+    ALL = KERNEL_MAPPING.keys.map { |k| CpuMitigations.new(k) }
+    DEFAULT = CpuMitigations.new(:auto)
+
+    def self.from_kernel_params(kernel_params)
+      log.info "kernel params #{kernel_params.inspect}"
+      param = kernel_params.parameter("mitigations")
+      log.info "mitigation param #{param.inspect}"
+      param = nil if param == false
+      reverse_mapping = KERNEL_MAPPING.invert
+      raise "Unknown mitigations value #{param.inspect}" if !reverse_mapping.key?(param)
```

Comment:
Um, where is this actually used? Is the exception handled somewhere?

I mean what happens when I boot with `mitigations=invalid_valued` boot parameter? Does it crash?

Reply:
yep, it is more like assertion that there is no unknown parameter. I am not fan of making it ConfigurationError, as we can be wrong and proposing it from scratch does not help as it failing in proposal.

Reply:
OK, in that case I'd prefer a more verbose message so the user knows what wrong and how to fix it. Something like:

`Unknown mitigations value #{param.inspect} in the kernel command line, supported values are: #{KERNEL_MAPPING.values.reject{|v| v.nil?}.map(&:inspect).join(", ")}.`

Reply:
`reject{|v| v.nil?}` is IIRC called `compact`